docs/src/vis.md

```diff
@@ -0,0 +1,82 @@
+# Visualization
+
+SparseMatrixColorings provides some internal utilities for visualization of matrix colorings via the un-exported function [`show_colors`](@ref).
+
+!!! warning
+    This function makes use of the [JuliaImages](https://juliaimages.org) ecosystem.
+    Using it requires loading at least [Colors.jl](https://github.com/JuliaGraphics/Colors.jl).
+    We recommend loading the full [Images.jl](https://github.com/JuliaImages/Images.jl) package for convenience, which includes Colors.jl.
+
+## Basic usage
+
+To obtain a visualization, simply call `show_colors` on a coloring result:
+
+```@example img
+using Images
+using SparseMatrixColorings, SparseArrays
+using SparseMatrixColorings: show_colors
+
+S = sparse([
+    0 0 1 1 0 1
+    1 0 0 0 1 0
+    0 1 0 0 1 0
+    0 1 1 0 0 0
+]);
+
+problem = ColoringProblem(; structure=:nonsymmetric, partition=:column)
+algo = GreedyColoringAlgorithm(; decompression=:direct)
+result = coloring(S, problem, algo)
+show_colors(result)
+```
+
+!!! tip "Terminal support"
+    Loading [ImageInTerminal.jl](https://github.com/JuliaImages/ImageInTerminal.jl) will allow you to show the output of `show_colors` within your terminal.
+    If you use VSCode's Julia REPL, the matrix will be displayed in the plot tab.
+
+## Customization
+
+The visualization can be customized via keyword arguments.
+The size of the matrix entries is defined by `scale`, while gaps between them are dictated by `pad`.
+We recommend using the [ColorSchemes.jl](https://github.com/JuliaGraphics/ColorSchemes.jl) catalogue to customize the `colorscheme`.
+Finally, a background color can be passed via the `background` keyword argument. To obtain transparent backgrounds, use the `RGBA` type.
+
+```@example img
+using ColorSchemes
+
+julia_colors = ColorSchemes.julia
+white = RGB(1, 1, 1)
+show_colors(result; colorscheme=julia_colors, background=white, scale=5, pad=1)
+```
+
+## Working with large matrices
+
+Let's demonstrate visualization of a larger random matrix:
+
+```@example img
+S = sprand(50, 50, 0.1) # sample sparse matrix
+
+problem = ColoringProblem(; structure=:nonsymmetric, partition=:column)
+algo = GreedyColoringAlgorithm(; decompression=:direct)
+result = coloring(S, problem, algo)
+show_colors(result; scale=5, pad=1)
+```
+
+Instead of the default `distinguishable_colors` from Colors.jl, one can subsample a continuous colorscheme from ColorSchemes.jl:
```

Comment:
@adrhill is there a way for the user to provide a naked colorscheme that gets automatically subsampled to the right number of colors?

Reply:
I guess that would require a dependence on ColorSchemes.jl. Their next release will include a `resample` function: https://juliagraphics.github.io/ColorSchemes.jl/dev/basics/#The-resample()-function

Reply:
let's leave it that way